### DISCUSSION TITLE
truffle compile error

### DISCUSSION THREAD

**Discussion by devdattanarawade97** [ConsenSys-archive/truffle#6227] [General Q&A]
as i run truffle compile command am getting following error : Error [ERR_REQUIRE_ESM]: require() of ES Module .truffle-config.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules. Instead either rename truffle-config.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in D:\DAPP\package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

**Comment by Venkat2552004** [comment]
Go to your `package.json`  file and change the ~` type : module`~ to `type : commonjs` and for all **`.js`** files in which you use `import()` instead of `require()`, change the file to have **`.mjs`** extension instead of .js extension

**Step - 1**

![image](https://github.com/trufflesuite/truffle/assets/109226829/17be015a-4fd0-4a2e-a07c-29f68fff5954)

**Step - 2**

![image](https://github.com/trufflesuite/truffle/assets/109226829/8c049eca-d156-4623-9101-56902ff9086f)
